contracts/multisig/src/contract/execute.rs

```diff
@@ -29,12 +29,14 @@ pub fn start_signing_session(
     let session_id = SIGNING_SESSION_COUNTER.update(
         deps.storage,
         |mut counter| -> Result<Uint64, ContractError> {
-            counter += Uint64::one();
+            counter = counter
+                .checked_add(Uint64::one())
+                .map_err(StdError::overflow)?;
             Ok(counter)
         },
     )?;
 
-    let expires_at = env.block.height + config.block_expiry;
+    let expires_at = env.block.height.saturating_add(config.block_expiry);
```

Comment:
in general, when the overflow was not caused by user input from the current transaction, then I lean towards saturation, the reason is that I prefer to let the protocol continue with some value rather than bricking the protocol. In this particular case I think it's more likely that the overflow is caused due to a misconfigured block_expiry (or a bug in the network?) and if that's the case then new signing sessions won't be produced until something is fixed

Reply:
a wrong configuration will lead to potentially wrong votes from this ampd instance, so the verifier could get penalized and worst case thrown out without being aware of what the problem was